src/api/query.rs

```diff
@@ -0,0 +1,402 @@
+use actix_web::{get, web, HttpResponse};
+use futures::TryStreamExt;
+use mongodb::{
+    bson::{doc, Document},
+    Client, Collection, IndexModel,
+};
+use std::collections::HashMap;
+
+use crate::{
+    api::util,
+    models::{query_models::*, response},
+};
+
+const DB_NAME: &str = "boom";
+
+// builds find options for mongo query
+pub fn build_options(
+    projection: Option<mongodb::bson::Document>,
+    kwargs: QueryKwargs,
+) -> mongodb::options::FindOptions {
+    let mut find_options = mongodb::options::FindOptions::default();
+    if kwargs.limit.is_some() {
+        find_options.limit = Some(kwargs.limit.unwrap());
+    }
+    if kwargs.skip.is_some() {
+        find_options.skip = Some(kwargs.skip.unwrap());
+    }
+    if kwargs.sort.is_some() {
+        find_options.sort = Some(kwargs.sort.unwrap());
+    }
+    if kwargs.max_time_ms.is_some() {
+        find_options.max_time = Some(std::time::Duration::from_millis(
+            kwargs.max_time_ms.unwrap(),
+        ));
+    }
+    if projection.is_some() {
+        find_options.projection = Some(projection.unwrap());
+    }
+
+    find_options
+}
+
+pub fn build_cone_search_filter(
+    mut filter: mongodb::bson::Document,
+    radec: (f64, f64),
+    mut radius: f64,
+    unit: Unit,
+) -> mongodb::bson::Document {
+    let ra = radec.0 - 180.0;
```

Comment:
not really needed to create these variables, can just do that `-180` when setting the value a couple of lines under that